ocaml/typing/typecore.ml

```diff
@@ -7477,22 +7532,20 @@ and type_argument ?explanation ?recarg env (mode : expected_mode) sarg
 and type_apply_arg env ~app_loc ~funct ~index ~position_and_mode ~partial_app (lbl, arg) =
   match arg with
   | Arg (Unknown_arg { sarg; ty_arg_mono; mode_arg; sort_arg }) ->
-      let mode, _ = Alloc.newvar_below mode_arg in
```

Comment:
If you don't remove this, then I think `mode_argument` doesn't need to return a tuple.

Reply:
I think I moved the `newvar_below` into `mode_argument` because they seem to be rather coupled.